### PR TITLE
[MaterializedView] fix mv use bugs

### PR DESCRIPTION
**Pull request by blackstar-baba**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes list:
1. can't `select * from mv`
2. can't `desc mv all`
3. can't `show partitions from mv`
4. can't insert data to mv from base table

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/alter/Alter.java [hunk 1]
```diff
@@ -175,7 +175,7 @@ public void processDropMaterializedView(DropMaterializedViewStmt stmt) throws Dd
                 }
             }
             if (table == null) {
-                throw new MetaNotFoundException("Materialized view " + stmt.getMvName() + " is not find");
+                throw new MetaNotFoundException("Materialized view " + stmt.getMvName() + "     is not find");
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/alter/Alter.java:178]: why add many spaces?

Comment (blackstar-baba) [reply fe/fe-core/src/main/java/com/starrocks/alter/Alter.java:178]: My fault，i will check all changes

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 20 / 26 (76.92%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/scheduler/MvTaskRunProcessor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/common/proc/TableProcDir.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|11|13|84.62%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/analysis/DescribeStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/transaction/TransactionStateListenerFactory.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/transaction/TransactionLogApplierFactory.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/common/proc/TableProcDir.java [hunk 2]
```diff
@@ -86,15 +86,15 @@ public ProcNodeInterface lookup(String entryName) throws AnalysisException {
         }
 
         if (entryName.equals(PARTITIONS)) {
-            if (table.getType() == TableType.OLAP) {
+            if (table.getType() == TableType.OLAP || table.getType() == TableType.MATERIALIZED_VIEW) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/TableProcDir.java:89]: I think we could add a method to unify the OLAP and MATERIALIZED_VIEW type.  such as isNativeTable?

Comment (blackstar-baba) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/TableProcDir.java]: ok

File: fe/fe-core/src/main/java/com/starrocks/transaction/TransactionStateListenerFactory.java [hunk 3]
```diff
@@ -11,7 +11,7 @@ public TransactionStateListener create(DatabaseTransactionMgr dbTxnMgr, Table ta
         if (table.isLakeTable()) {
             return new LakeTableTxnStateListener(dbTxnMgr, (LakeTable) table);
         }
-        if (table.isOlapTable()) {
+        if (table.isOlapTable() || table.isMaterializedView()) {
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/transaction/TransactionStateListenerFactory.java:14]:
```suggestion
        if (table.isNativeTable()) {
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 21 / 26 (80.77%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/MvTaskRunProcessor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/common/proc/TableProcDir.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|11|13|84.62%|
|:large_blue_circle:|com/starrocks/catalog/Table.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/DescribeStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/transaction/TransactionStateListenerFactory.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/common/proc/PartitionsProcDir.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/transaction/TransactionLogApplierFactory.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/scheduler/TaskBuilder.java|1|1|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Pull request merged**